### PR TITLE
Fix: Allow changed() function to work properly with collection properties of type object

### PR DESCRIPTION
**Pull request by andreialecu**
changed('prop') in event scripts would return true even if the object was not changed. This fixes it by using underscore's deep object comparison functionality.

### CODE REVIEW

File: lib/resources/collection/index.js [hunk 1]
```diff
@@ -448,7 +448,7 @@ Collection.prototype.save = function (ctx, fn) {
 
   domain.changed =  function (property) {
```
Comment (ericfong) [inline lib/resources/collection/index.js]: Shall we 
```js
                    var args = Array.prototype.slice.call(arguments);
                    for (var i = 0, ii = args.length;  i < ii; i++) {

// The original code...

                    }
                    return false;
```

So that we can call ```changed('fieldA', 'fieldB')``` to detect ```fieldA``` OR ```fieldB``` is changed

Why OR?
In one-to-many or many-to-many relationship, actions are more likely to be triggered by sub-set of fields on a item.

Comment (andreialecu) [inline lib/resources/collection/index.js]: That would be nice I guess, it's a feature that doesn't break anything. I can update the PR with it if you want.

Comment (ericfong) [inline lib/resources/collection/index.js]: I go for OR case. 

I would like to know anyone will go for AND case and reason for that.

Comment (andreialecu) [inline lib/resources/collection/index.js]: I still think it's somewhat unintuitive and can lead to issues.

How about changedAny('fieldA', 'fieldB') and changedAll('fieldA', 'fieldB') as two separate methods? It would be just as simple to implement.

I can think about things like custom logic for notifying an admin if ALL the specified fields are changed, but I can definitely see the ANY case being just as useful.

As an example, I could have:
```
if (changedAll('availabilityStatus', 'availabilityUntil')) {
  ..logic
} else {
  cancel('both availabilityStatus and availabilityUntil to be updated simultaneously');
}
```

Comment (ericfong) [inline lib/resources/collection/index.js]: changedAny & changedAll can be in another PR. Then, we can keep this PR unchanged.

**Comment by ericfong** [comment]
In case anyone consider performance, 
according to https://github.com/jashkenas/underscore/blob/master/underscore.js#L1083
Should not have bit performance different

**Comment by ericfong** [comment]
landed as 46f518dc876e723c6c87c70a87b6ef7278b6dab4
Thanks for your contribution!
